nat-lab/tests/test_notification_center.py

```diff
@@ -0,0 +1,120 @@
+import json
+from asyncio import sleep
+from contextlib import AsyncExitStack
+from dataclasses import dataclass
+from utils.connection_util import ConnectionTag, new_connection_by_tag
+from uuid import UUID
+
+
+@dataclass
+class MachineResponse:
+    identifier: str
+    public_key: str
+    hostname: str
+    os: str
+    os_version: str
+    device_type: str
+    nickname: str
+    ip_addresses: list[str]
+    discovery_key: str
+    relay_address: str
+    traffic_routing_supported: bool
+
+
+def verify_uuid(uuid_to_test, version=4):
+    try:
+        uuid_obj = UUID(uuid_to_test, version=version)
+    except ValueError:
+        assert False, "Not a valid UUID"
+    assert str(uuid_obj) == uuid_to_test, "Not a valid UUID"
+
+
+def verify_mqtt_payload(payload):
+    payload_json = json.loads(payload)
+
+    verify_uuid(payload_json["message"]["data"]["metadata"]["message_id"])
+    affected_machines_list = payload_json["message"]["data"]["event"]["attributes"][
+        "affected_machines"
+    ]
+    for machine_uuid in affected_machines_list:
+        verify_uuid(machine_uuid)
+
+
+async def run_mqtt_listener(exit_stack, connection):
+    # Start listening for mqtt message
+    mqtt_process = await exit_stack.enter_async_context(
+        connection.create_process(["python3", "/opt/bin/mqtt_listener.py"]).run()
+    )
+
+    # Allows mqtt process to run
+    await sleep(0.1)
+
+    return mqtt_process
+
+
+async def test_nc_register():
+    async with AsyncExitStack() as exit_stack:
+        # Setup connections
+        connection_alpha = await exit_stack.enter_async_context(
+            new_connection_by_tag(ConnectionTag.DOCKER_CONE_CLIENT_1)
+        )
+        connection_beta = await exit_stack.enter_async_context(
+            new_connection_by_tag(ConnectionTag.DOCKER_CONE_CLIENT_2)
+        )
+
+        mqtt_process = await run_mqtt_listener(exit_stack, connection_alpha)
```

Comment:
I think you should be able to establish MQTT connection directly from this test instead of running it in separate process. I guess it does not really have to be run in one of the `DOCKER_CONE_CLIENT_X` networks.